sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Watch.java

```diff
@@ -955,11 +961,15 @@ public ProcessContinuation process(
         return stop();
       }
 
-      if (BoundedWindow.TIMESTAMP_MAX_VALUE.equals(newResults.getWatermark())) {
+      if (BoundedWindow.TIMESTAMP_MAX_VALUE.equals(computedWatermark)) {
         LOG.info("{} - will stop polling, reached max timestamp.", c.element());
         return stop();
       }
 
+      if (computedWatermark != null) {
```

Comment:
We want to set the `watermark` for `watermarkEstimator` even when `stop()` is returned, right?

Reply:
Since the restriction is done, wouldn't the watermark not be part of the `min` set of remaining restrictions being processed?

Reply:
That's true.